.github/workflows/test.yml

```diff
@@ -1,27 +1,29 @@
 name: Run Lint and Tests
 
-on: push
+on:
+    pull_request:
+        branches: [main]
 
 jobs:
     build:
         runs-on: ubuntu-latest
         strategy:
             matrix:
                 os: [ubuntu-latest, macOS-latest, windows-latest]
-                node-version: [10.x, 12.x, 13.x]
+                node-version: [10.x, 18.x, 20.x]
```

Comment:
I think we can probably drop 10.x

Reply:
It should hopefully not be in use anywhere, but I figured officially dropping support is a breaking change, and wanted to avoid that here.